### DISCUSSION TITLE
Issues building firmware and getting log data.

### DISCUSSION THREAD

**Discussion by santiagorg2401** [USC-ACTLab/crazyswarm#433] [Q&A]
Hi, I hope you're all doing well,
Lately I've been having 2 problems, the first one is building the Crazyflie firmware (crazyflie-firmware), I always get the following error:

```
CLEAN_VERSION
make[1]: *** No rule to make target 'list.o', needed by 'cf2.elf'.  Stop.
Makefile:375: recipe for target 'build' failed
make: *** [build] Error 2
```

So, I re-installed everything again, followed the instructions given in the docs and still get the same error (even with no modifications at all).

The second problem is getting the stateEstimate.x log data, when running the master node I got:

```
[ INFO] [1626887030.764685806]: [cf1] reqLogTOC: 0.003347 s
terminate called after throwing an instance of 'std::runtime_error'
  what():  Could not find stateEstimate.x in log toc!
[crazyswarm_server-2] process has died [pid 22408, exit code -6, cmd /home/santiagorg2401/crazyflie/obstacleAvoidance1.0/crazyswarm/ros_ws/devel/lib/crazyswarm/crazyswarm_server __name:=crazyswarm_server __log:=/home/santiagorg2401/.ros/log/9e6467a0-ea45-11eb-8073-181deab367da/crazyswarm_server-2.log].
log file: /home/santiagorg2401/.ros/log/9e6467a0-ea45-11eb-8073-181deab367da/crazyswarm_server-2*.log
[ INFO] [1626887030.764685806]: [cf1] reqLogTOC: 0.003347 s
terminate called after throwing an instance of 'std::runtime_error'
  what():  Could not find stateEstimate.x in log toc!
[crazyswarm_server-2] process has died [pid 22408, exit code -6, cmd /home/santiagorg2401/crazyflie/obstacleAvoidance1.0/crazyswarm/ros_ws/devel/lib/crazyswarm/crazyswarm_server __name:=crazyswarm_server __log:=/home/santiagorg2401/.ros/log/9e6467a0-ea45-11eb-8073-181deab367da/crazyswarm_server-2.log].
log file: /home/santiagorg2401/.ros/log/9e6467a0-ea45-11eb-8073-181deab367da/crazyswarm_server-2*.log
```

The funny thing is that it only happens with stateEstimate.x, with y and z it doesn't have any trouble.

A couple of months ago I didn't get any of these errors, then I re-installed Ubuntu and  I had the issues I had the issues with the last version of the crazyswarm project.

Thank you for your time.

**Comment by whoenig** [comment]
For your first problem: The latest version does not build the Crazyflie firmware anymore, since it assumes you installed an official firmware directly. Are you running an older version of the Crazyswarm?

For your second problem: There are quite many reports with similar issues, but we have not found a root cause yet as we cannot reproduce the problem. Make sure that you remove the cache files (`rm ~/.ros/*.csv`) and try again. If that doesn't help, please let us know if `rosrun crazyflie_tools listLogVariables` contains `stateEstimate.x` or not.

**Comment by santiagorg2401** [comment]
1. Got it, I didn't know it doesn't build anymore.

2. I removed the cache files and I can get the stateEstimate.x, but, now I can't get range.zrange (before removing the cache files I could get it).

**Reply by whoenig** [reply]
Could you try setting `force_no_cache` to true in the launch file? Also, what is the output of `rosrun crazyflie_tools listLogVariables --uri <YOUR-URI>`?

**Reply by santiagorg2401** [reply]
I flashed the Crazyflie with the 2021.01 firmware and now it seems to work, although range.zrange doesn't appear in  listLogVariables, I can get the data.

listLogVariables output:
ctrlINDI.u_act_dyn.p (float)
gyro.yRaw (int16)
gyro.zRaw (int16)
gyro.xVariance (float)
gyro.yVariance (float)
gyro.zVariance (float)
pwm.m1_pwm (uint32)
pwm.m2_pwm (uint32)
pwm.m3_pwm (uint32)
pwm.m4_pwm (uint32)
crtp.rxRate (uint16)
crtp.txRate (uint16)
pm.vbat (float)
pm.vbatMV (uint16)
pm.extVbat (float)
pm.extVbatMV (uint16)
pm.extCurr (float)
pm.chargeCurrent (float)
pm.state (int8)
pm.batteryLevel (uint8)
radio.rssi (uint8)
radio.isConnected (uint8)
sys.canfly (int8)
sys.armed (int8)
sitAw.FFAccWZDetected (uint8)
sitAw.ARDetected (uint8)
sitAw.TuDetected (uint8)
kalman.varY (float)
extrx.ch1 (uint16)
extrx.ch2 (uint16)
extrx.ch3 (uint16)
extrx.thrust (uint16)
extrx.roll (float)
extrx.pitch (float)
extrx.yaw (float)
memTst.errCntW (uint32)
range.front (uint16)
range.back (uint16)
range.up (uint16)
range.left (uint16)
range.right (uint16)
kalman.rtBaro (float)
ext_pos.X (float)
ext_pos.Y (float)
ext_pos.Z (float)
locSrvZ.tick (uint16)
kalman_states.oy (float)
pid_attitude.roll_outI (float)
pid_attitude.roll_outD (float)
pid_attitude.pitch_outP (float)
pid_attitude.pitch_outI (float)
pid_attitude.pitch_outD (float)
pid_attitude.yaw_outP (float)
pid_attitude.yaw_outI (float)
pid_attitude.yaw_outD (float)
pid_rate.roll_outP (float)
pid_rate.roll_outI (float)
pid_rate.roll_outD (float)
pid_rate.pitch_outP (float)
pid_rate.pitch_outI (float)
pid_rate.pitch_outD (float)
pid_rate.yaw_outP (float)
pid_rate.yaw_outI (float)
pid_rate.yaw_outD (float)
sensfusion6.qw (float)
sensfusion6.qx (float)
sensfusion6.qy (float)
sensfusion6.qz (float)
sensfusion6.gravityX (float)
sensfusion6.gravityY (float)
sensfusion6.gravityZ (float)
sensfusion6.accZbase (float)
ranging.distance3 (float)
sensfusion6.isCalibrated (uint8)
acc.x (float)
acc.y (float)
acc.z (float)
baro.asl (float)
baro.temp (float)
ranging.pressure2 (float)
controller.ctr_yaw (int16)
ctrltarget.x (float)
ctrltarget.y (float)
ranging.pressure6 (float)
ctrltarget.vx (float)
ctrltarget.vy (float)
ctrltarget.vz (float)
ctrltarget.ax (float)
twr.rangingPerSec1 (uint8)
twr.rangingSuccessRate2 (uint8)
twr.rangingPerSec2 (uint8)
ctrltarget.pitch (float)
ctrltarget.yaw (float)
ctrltargetZ.x (int16)
ctrltargetZ.y (int16)
ctrltargetZ.z (int16)
ctrltargetZ.vx (int16)
ctrltargetZ.vy (int16)
ctrltargetZ.vz (int16)
tdoa2.d1-2 (float)
ctrltargetZ.ay (int16)
ctrltargetZ.az (int16)
gyro.x (float)
gyro.y (float)
gyro.z (float)
health.motorVarXM1 (float)
health.motorVarYM1 (float)
health.motorVarXM2 (float)
health.motorVarYM2 (float)
health.motorVarXM3 (float)
health.motorVarYM3 (float)
health.motorVarXM4 (float)
health.motorVarYM4 (float)
health.motorPass (uint8)
health.motorTestCount (uint16)
health.checkStops (uint8)
mag.x (float)
mag.y (float)
tdoa2.dist4-5 (uint16)
stabilizer.roll (float)
stabilizer.pitch (float)
tdoaEngine.stRx (float)
stabilizer.thrust (float)
stabilizer.rtStab (float)
stabilizer.intToOut (uint32)
stateEstimate.x (float)
stateEstimate.y (float)
stateEstimate.z (float)
stateEstimate.vx (float)
stateEstimate.vy (float)
stateEstimate.vz (float)
stateEstimate.ax (float)
stateEstimate.ay (float)
stateEstimate.az (float)
stateEstimate.roll (float)
stateEstimate.pitch (float)
stateEstimate.yaw (float)
stateEstimate.qx (float)
stateEstimate.qy (float)
stateEstimate.qz (float)
stateEstimate.qw (float)
stateEstimateZ.x (int16)
stateEstimateZ.y (int16)
stateEstimateZ.z (int16)
stateEstimateZ.vx (int16)
stateEstimateZ.vy (int16)
stateEstimateZ.vz (int16)
stateEstimateZ.ax (int16)
stateEstimateZ.ay (int16)
stateEstimateZ.az (int16)
stateEstimateZ.quat (uint32)
stateEstimateZ.rateRoll (int16)
stateEstimateZ.ratePitch (int16)
stateEstimateZ.rateYaw (int16)
posEstAlt.estimatedZ (float)
posEstAlt.estVZ (float)
posEstAlt.velocityZ (float)
posCtl.targetVX (float)
posCtl.targetVY (float)
posCtl.targetVZ (float)
posCtl.targetX (float)
posCtl.targetY (float)
posCtl.targetZ (float)
posCtl.Xp (float)
lighthouse.angle1y (float)
posCtl.Xd (float)
lighthouse.angle0y_1 (float)
posCtl.Yi (float)
posCtl.Yd (float)
posCtl.Zp (float)
posCtl.Zi (float)
posCtl.Zd (float)
posCtl.VXp (float)
posCtl.VXi (float)
lighthouse.angle0y_3 (float)
posCtl.VZp (float)
posCtl.VZi (float)
posCtl.VZd (float)
posCtrlIndi.gyr_p (float)
posCtrlIndi.gyr_q (float)
posCtrlIndi.gyr_r (float)
posCtrlIndi.posRef_x (float)
posCtrlIndi.posRef_y (float)
posCtrlIndi.posRef_z (float)
posCtrlIndi.velS_x (float)
posCtrlIndi.velS_y (float)
posCtrlIndi.velS_z (float)
posCtrlIndi.velRef_x (float)
posCtrlIndi.velRef_y (float)
posCtrlIndi.velRef_z (float)
posCtrlIndi.angS_roll (float)
posCtrlIndi.angS_pitch (float)
posCtrlIndi.angS_yaw (float)
posCtrlIndi.angF_roll (float)
posCtrlIndi.angF_pitch (float)
posCtrlIndi.angF_yaw (float)
posCtrlIndi.accRef_x (float)
posCtrlIndi.accRef_y (float)
posCtrlIndi.accRef_z (float)
posCtrlIndi.accS_x (float)
posCtrlIndi.accS_y (float)
posCtrlIndi.accS_z (float)
posCtrlIndi.accF_x (float)
posCtrlIndi.accF_y (float)
posCtrlIndi.accF_z (float)
posCtrlIndi.accFT_x (float)
posCtrlIndi.accFT_y (float)
posCtrlIndi.accFT_z (float)
posCtrlIndi.accErr_x (float)
posCtrlIndi.accErr_y (float)
posCtrlIndi.accErr_z (float)
posCtrlIndi.phi_tilde (float)
posCtrlIndi.theta_tilde (float)
posCtrlIndi.T_tilde (float)
posCtrlIndi.T_inner (float)
posCtrlIndi.T_inner_f (float)
posCtrlIndi.T_incremented (float)
posCtrlIndi.cmd_phi (float)
posCtrlIndi.cmd_theta (float)
controller.cmd_thrust (float)
controller.cmd_roll (float)
controller.cmd_pitch (float)
controller.cmd_yaw (float)
controller.r_roll (float)
controller.r_pitch (float)
controller.r_yaw (float)
controller.accelz (float)
controller.actuatorThrust (float)
controller.roll (float)
controller.pitch (float)
controller.yaw (float)
controller.rollRate (float)
controller.pitchRate (float)
controller.yawRate (float)
ctrlMel.cmd_thrust (float)
ctrlMel.cmd_roll (float)
ctrlMel.cmd_pitch (float)
ctrlMel.cmd_yaw (float)
ctrlMel.r_roll (float)
ctrlMel.r_pitch (float)
ctrlMel.r_yaw (float)
ctrlMel.accelz (float)
ctrlMel.zdx (float)
ctrlMel.zdy (float)
ctrlMel.zdz (float)
ctrlMel.i_err_x (float)
ctrlMel.i_err_y (float)
ctrlMel.i_err_z (float)
ctrlINDI.cmd_thrust (float)
ctrlINDI.cmd_roll (float)
ctrlINDI.cmd_pitch (float)
ctrlINDI.cmd_yaw (float)
ctrlINDI.r_roll (float)
ctrlINDI.r_pitch (float)
ctrlINDI.r_yaw (float)
ctrlINDI.accelz (float)
ctrlINDI.u_act_dyn.p (float)
ctrlINDI.u_act_dyn.q (float)
ctrlINDI.u_act_dyn.r (float)
ctrlINDI.du.p (float)
ctrlINDI.du.q (float)
ctrlINDI.du.r (float)
ctrlINDI.ang_accel_ref.p (float)
ctrlINDI.ang_accel_ref.q (float)
ctrlINDI.ang_accel_ref.r (float)
ctrlINDI.rate_d[0] (float)
ctrlINDI.rate_d[1] (float)
ctrlINDI.rate_d[2] (float)
motor.m1 (uint32)
motor.m2 (uint32)
motor.m3 (uint32)
motor.m4 (uint32)
kalman.inFlight (uint8)
kalman.stateX (float)
kalman.stateY (float)
kalman.stateZ (float)
kalman.statePX (float)
kalman.statePY (float)
kalman.statePZ (float)
kalman.stateD0 (float)
kalman.stateD1 (float)
kalman.stateD2 (float)
kalman.varX (float)
kalman.varY (float)
kalman.varZ (float)
kalman.varPX (float)
kalman.varPY (float)
kalman.varPZ (float)
kalman.varD0 (float)
kalman.varD1 (float)
kalman.varD2 (float)
kalman.q0 (float)
kalman.q1 (float)
kalman.q2 (float)
kalman.q3 (float)
kalman.rtUpdate (float)
kalman.rtPred (float)
kalman.rtBaro (float)
kalman.rtFinal (float)
kalman.rtApnd (float)
kalman.rtRej (float)
kalman_states.ox (float)
kalman_states.oy (float)
kalman_states.vx (float)
kalman_states.vy (float)
kalman_pred.predNX (float)
kalman_pred.predNY (float)
kalman_pred.measNX (float)
kalman_pred.measNY (float)
outlierf.lhWin (int32)
colAv.latency (uint32)
ring.fadeTime (float)
gps.lat (int32)
gps.lon (int32)
gps.hMSL (float)
gps.hAcc (float)
gps.nsat (int32)
gps.fix (int32)
usd.spiWrBps (float)
usd.spiReBps (float)
usd.fatWrBps (float)
loco.mode (uint8)
loco.spiWr (float)
loco.spiRe (float)
ranging.state (uint16)
ranging.distance0 (float)
ranging.distance1 (float)
ranging.distance2 (float)
ranging.distance3 (float)
ranging.distance4 (float)
ranging.distance5 (float)
ranging.distance6 (float)
ranging.distance7 (float)
ranging.pressure0 (float)
ranging.pressure1 (float)
ranging.pressure2 (float)
ranging.pressure3 (float)
ranging.pressure4 (float)
ranging.pressure5 (float)
ranging.pressure6 (float)
ranging.pressure7 (float)
twr.rangingSuccessRate0 (uint8)
twr.rangingPerSec0 (uint8)
twr.rangingSuccessRate1 (uint8)
twr.rangingPerSec1 (uint8)
twr.rangingSuccessRate2 (uint8)
twr.rangingPerSec2 (uint8)
twr.rangingSuccessRate3 (uint8)
twr.rangingPerSec3 (uint8)
twr.rangingSuccessRate4 (uint8)
twr.rangingPerSec4 (uint8)
twr.rangingSuccessRate5 (uint8)
twr.rangingPerSec5 (uint8)
tdoa2.d7-0 (float)
tdoa2.d0-1 (float)
tdoa2.d1-2 (float)
tdoa2.d2-3 (float)
tdoa2.d3-4 (float)
tdoa2.d4-5 (float)
tdoa2.d5-6 (float)
tdoa2.d6-7 (float)
tdoa2.cc0 (float)
tdoa2.cc1 (float)
tdoa2.cc2 (float)
tdoa2.cc3 (float)
tdoa2.cc4 (float)
tdoa2.cc5 (float)
tdoa2.cc6 (float)
tdoa2.cc7 (float)
tdoa2.dist7-0 (uint16)
tdoa2.dist0-1 (uint16)
tdoa2.dist1-2 (uint16)
tdoa2.dist2-3 (uint16)
tdoa2.dist3-4 (uint16)
tdoa2.dist4-5 (uint16)
tdoa2.dist5-6 (uint16)
tdoa2.dist6-7 (uint16)
tdoaEngine.stRx (float)
tdoaEngine.stEst (float)
tdoaEngine.stTime (float)
tdoaEngine.stFound (float)
tdoaEngine.stCc (float)
tdoaEngine.stHit (float)
tdoaEngine.stMiss (float)
tdoaEngine.cc (float)
tdoaEngine.tof (uint16)
tdoaEngine.tdoa (float)
outlierf.bucket0 (int32)
outlierf.bucket1 (int32)
outlierf.bucket2 (int32)
outlierf.bucket3 (int32)
outlierf.bucket4 (int32)
outlierf.accLev (float)
outlierf.errD (float)
motion.motion (uint8)
motion.deltaX (int16)
motion.deltaY (int16)
motion.shutter (uint16)
motion.maxRaw (uint8)
motion.minRaw (uint8)
motion.Rawsum (uint8)
motion.outlierCount (uint8)
motion.squal (uint8)
motion.std (float)
oa.front (uint16)
oa.back (uint16)
oa.up (uint16)
oa.left (uint16)
oa.right (uint16)
activeMarker.btSns (uint8)
activeMarker.i2cOk (uint8)
aideck.receivebyte (uint8)
lighthouse.validAngles (uint8)
lighthouse.rawAngle0x (float)
lighthouse.rawAngle0y (float)
lighthouse.rawAngle1x (float)
lighthouse.rawAngle1y (float)
lighthouse.angle0x (float)
lighthouse.angle0y (float)
lighthouse.angle1x (float)
lighthouse.angle1y (float)
lighthouse.angle0x_1 (float)
lighthouse.angle0y_1 (float)
lighthouse.angle1x_1 (float)
lighthouse.angle1y_1 (float)
lighthouse.angle0x_2 (float)
lighthouse.angle0y_2 (float)
lighthouse.angle1x_2 (float)
lighthouse.angle1y_2 (float)
lighthouse.angle0x_3 (float)
lighthouse.angle0y_3 (float)
lighthouse.angle1x_3 (float)
lighthouse.angle1y_3 (float)
lighthouse.rawAngle0xlh2 (float)
lighthouse.rawAngle0ylh2 (float)
lighthouse.rawAngle1xlh2 (float)
lighthouse.rawAngle1ylh2 (float)
lighthouse.angle0x_0lh2 (float)
lighthouse.angle0y_0lh2 (float)
lighthouse.angle1x_0lh2 (float)
lighthouse.angle1y_0lh2 (float)
lighthouse.serRt (float)
lighthouse.frmRt (float)
lighthouse.cycleRt (float)
lighthouse.bs0Rt (float)
lighthouse.bs1Rt (float)
lighthouse.width0 (uint16)
lighthouse.width1 (uint16)
lighthouse.width2 (uint16)
lighthouse.width3 (uint16)
lighthouse.comSync (uint8)
lighthouse.posRt (float)
lighthouse.estBs0Rt (float)
lighthouse.estBs1Rt (float)
lighthouse.x (float)
lighthouse.y (float)
lighthouse.z (float)
lighthouse.delta (float)
